utils/metric_database.py

```diff
@@ -0,0 +1,315 @@
+            Optional[List[Tuple[Any, ...]]]: a list of tuples representing the query results if 'fetch' is True, None otherwise.
+        """
+        result = None
+
+        try:
+            conn = self.get_conn()
+
+            with conn.cursor() as cursor:
+                cursor.execute(query)
+                if fetch:
+                    result = cursor.fetchall()
+
+            conn.commit()
+        except Exception as e:
+            logging.error(f"{e} error: Unable to execute the query.")
+            conn.rollback()  # roll back transaction on error
+        finally:
+            if self.conn_pool:
+                self.conn_pool.putconn(conn)  # return connection back to pool
+
+        return result
+
+    def check_relation_existence(self, relation_name: str) -> bool:
+        """This function checks whether the readability relation exists.
+
+        Args:
+            relation_name (str): name of the relation to check for.
+
+        Returns:
+            bool: returns True if exists, False otherwise
+        """
+        result = self.execute_query(
+            SQLQueries.relation_existence_query(relation_name), fetch=True
+        )
+        if result:  # mypy
+            return bool(result[0][0])
+
+        return False
+
+    def create_relation(self, relation_name: str) -> None:
+        """This function creates a relation."""
+        query_map = {
+            "Readability": SQLQueries.create_readability_relation_query(),
+            "EmbeddingDrift": SQLQueries.create_embedding_drift_relation_query(),
+        }
+        self.execute_query(str(query_map.get(relation_name)))
+
+    def insert_readability_data(self, score: float) -> None:
+        """This function insert a row of data into to readability relation.
+
+        Args:
+            score (float): the readability score computed.
+        """
+        self.execute_query(SQLQueries.insert_readability_data(score))
+
+    def insert_embedding_drift_data(
+        self, data: Dict[str, Union[str, float, bool]]
+    ) -> None:
+        """This function insert a row of data into to EmbeddingDrift relation.
+
+        Args:
+            data (Dict[str, Union[str, float, bool]]): a dictionary containing the embedding drift data to be inserted to the relation.
+        """
+        self.execute_query(SQLQueries.insert_embedding_drift_data(data))
+
+    def query_relation(self, relation_name: str) -> List[Tuple[Any, ...]]:
+        """This function queries a specific relation in the database, based on the provided relation name.
+
+        Args:
+            relation_name (str): the name of the relation in database
+
+        Returns:
+            List[Tuple[Any, ...]]: a list of tuples representing the data rows from the queried relation.
+        """
+        result = self.execute_query(
+            SQLQueries.get_data_from_relation(relation_name), fetch=True
+        )
+
+        return result if result else []  # mypy will complain if just return result
```

Comment:
Is the solution to the mypy problems to fix the type of the variable when it's created? E.g., ```result: <type> = self...```